apisix/plugin.lua

```diff
@@ -503,6 +503,11 @@ local function merge_consumer_route(route_conf, consumer_conf)
         end
 
```

Comment:
we can use `if new_route_conf.value.plugins[name] == nil then conf._from_consumer = true`?

Reply:
tried, Test 2 will fail. We need to merge the `new_route_conf.value.plugins["proxy-rewrite"],  it will not be nil.

Reply:
What about this change?
```
diff --git apisix/plugin.lua apisix/plugin.lua
index d66ec045..190b0833 100644
--- apisix/plugin.lua
+++ apisix/plugin.lua
@@ -502,12 +502,11 @@ local function merge_consumer_route(route_conf, consumer_conf)
             new_route_conf.value.plugins = {}
         end

-        new_route_conf.value.plugins[name] = conf
-
-        if (route_conf and route_conf.value and route_conf.value.plugins)
-                and not route_conf.value.plugins[name] then
-            new_route_conf.value.plugins[name]["_from_consumer"] = true
+        if new_route_conf.value.plugins[name] == nil then
+            conf._from_consumer = true
         end
+
+        new_route_conf.value.plugins[name] = conf
     end

     core.log.info("merged conf : ", core.json.delay_encode(new_route_conf))
```

Reply:
yes, update